specs/phase0/p2p-interface.md

```diff
@@ -604,6 +607,50 @@ Nonetheless, ENRs MUST carry a generic `eth2` key with nil value, denoting that
 
 #### Mainnet
 
+##### `eth2` field
+
+ENRs MUST carry a generic `eth2` key with an 16-byte value of the node's current fork version, next fork version, and next fork epoch to ensure connections are made with peers on the intended eth2 network.
+
+| Key          | Value               |
+|:-------------|:--------------------|
+| `eth2`       | SSZ `ENRForkID`        |
+
+First we define `current_fork` as the following SSZ encoded object
+
+```
+(
+    current_fork_version: Version
+    genesis_validators_root: Root
+)
+```
+
+where
+
+* `current_fork_version` is the fork version at the node's current epoch defined by the wall-clock time (not necessarily the epoch to which the node is sync)
+* `genesis_validators_root` is the static `Root` found in `state.genesis_validators_root`
+
+Specifically, the value of the `eth2` key MUST be the following SSZ encoded object (`ENRForkID`)
+
+```
+(
+    current_fork_digest: Bytes4
+    next_fork_version: Version
+    next_fork_epoch: Epoch
+)
+```
+
+where the fields of `ENRForkID` are defined as
+
+* `current_fork_digest` is `hash_tree_root(current_fork)[:4]`
```

Comment:
Should we make a fork-digest data-type? To make it obviously different from `Version`, given that both are a bytes4? Can also do cosmetics later if necessary.

Reply:
Maybe so. I'll see if I can quickly, cleanly add this in